apps/aefate/src/aefa_fate_op.erl

```diff
@@ -1247,6 +1261,134 @@ spend(Arg0, Arg1, ES0) ->
             aefa_fate:abort({value_does_not_match_type, Other, address}, ES2)
     end.
 
+create(Arg0, Arg1, Arg2, ES0) ->
+    ?AVAILABLE_FROM(?VM_FATE_SOPHIA_2, ES0),
+    {[?FATE_CONTRACT_BYTEARRAY(Code), InitArgsTypes, Value], ES1} =
+        get_op_args([Arg0, Arg1, Arg2], ES0),
+    create_common(Code, InitArgsTypes, Value, no_gas_cap, ES1).
+
+create_common(Code, InitArgsTypes, Value, GasCap, ES0) ->
+    {ContractPK, ES1} = create_contract({new, Code}, Value, ES0),
+    ReplaceInitResult
+        = fun(ES0_) ->
+                  %% init returns unit, so we get rid of it
+                  {{tuple, {}}, ES1_} = get_op_arg({stack, 0}, ES0_),
+                  write({stack, 0}, ?FATE_CONTRACT(ContractPK), ES1_)
+          end,
+    {ok, ES2} = remote_call_common(
+           ?FATE_CONTRACT(ContractPK),
+           init,
+           InitArgsTypes,
+           ?FATE_TYPEREP({tuple, []}),
+           0,
+           GasCap,
+           unprotected,
+           ReplaceInitResult,
+           ES1),
+    {jump, 0, ES2}.
+
+clone(Arg0, Arg1, Arg2, Arg3, ES0) ->
+    ?AVAILABLE_FROM(?VM_FATE_SOPHIA_2, ES0),
+    {[?FATE_CONTRACT(CloneePK), InitArgsTypes, Value, Prot], ES1} =
+        get_op_args([Arg0, Arg1, Arg2, Arg3], ES0),
+    clone_common(CloneePK, InitArgsTypes, Value, no_gas_cap, Prot, ES1).
+
+clone_g(Arg0, Arg1, Arg2, Arg3, Arg4, ES0) ->
+    ?AVAILABLE_FROM(?VM_FATE_SOPHIA_2, ES0),
+    {[?FATE_CONTRACT(CloneePK), InitArgsTypes, Value, GasCap, Prot], ES1} =
+        get_op_args([Arg0, Arg1, Arg2, Arg3, Arg4], ES0),
+    clone_common(CloneePK, InitArgsTypes, Value, {gas_cap, GasCap}, Prot, ES1).
+
+clone_common(CloneePK, InitArgsTypes, Value, GasCap, Prot, ES0) ->
```

Comment:
`clone_common` and `create_common` are sufficiently similar to merit a refactoring it could take `{new, Code}` or `{clone, PK}` as argument and set `Protected` to `unprotected` in the create case.

Reply:
done